### PR TITLE
Extend `Wasm` trait & generalize `App::store_code`

### PR DESCRIPTION
**Pull request by KirilMihaylov**
Reopen of #50 after it was closed due to an accident.

### CODE REVIEW

**Comment by DariuszDepta** [comment]
Hi @KirilMihaylov!

Would you take a look again at the current implementation?
Current version of `WasmKeeper::store_code` is unlikely to fail until you would like to store quite a big number of tested contracts, so returning `u64` wrapped in Result is - in my opinion - not needed.
Having a Result would suggest to the user, that in some circumstances (easy to achieve), the `App::store_code` function could fail. The need of adding `unwrap()` in all test cases is just additional typing work...

Let me know what you think.

Thanks in advance.

**Comment by KirilMihaylov** [comment]
Hi @DariuszDepta.

To me it didn't seem idiomatic to do it in such a way, thus I used the `Result` type, but the code base isn't mine to choose standards of writing code, so it is I who should comply.

In case there is anything else that concerns you, feel free to `@` me.

**Comment by DariuszDepta** [comment]
Hi @KirilMihaylov, I guess the merged PR #74 addresses the needs from this PR. Could you possibly resolve conflicts then we could merge your PR?

**Comment by KirilMihaylov** [comment]
Hi @DariuszDepta! I resolved the conflicts. Feel free to merge it, if it looks good.

**Comment by DariuszDepta** [comment]
Hi @KirilMihaylov, looks good, just I guess your local main branch must be refreshed, because the code on origin/main in file `app.rs` was changed exactly in the same place that yours...

File: src/app.rs [hunk 1]
```diff
@@ -701,7 +701,34 @@ where
...
+    ApiT: Api,
+    StorageT: Storage,
+    CustomT: Module,
+    StakingT: Staking,
+    DistrT: Distribution,
+    IbcT: Ibc,
+    GovT: Gov,
+    CustomT::ExecT: Clone + Debug + PartialEq + JsonSchema + DeserializeOwned + 'static,
+    CustomT::QueryT: CustomQuery + DeserializeOwned + 'static,
+{
```
Comment (DariuszDepta) [inline src/app.rs:731]: This code is newer on origin/main. I guess refreshing/updating your fork should help.

Comment (KirilMihaylov) [reply src/app.rs:731]: Okay, I'll do a rebase and compare the changes.

**Comment by KirilMihaylov** [comment]
Hi @DariuszDepta. I suppose I'll just close the PR it seems you moved everything in #74 and it's just white spaces now.

**Comment by DariuszDepta** [comment]
Anyway thanks a lot! Without your work and suggestions there would be no #74 PR! Thanks!

**Comment by DariuszDepta** [comment]
Implemented in #74.

**Pull request closed without merging**